### PR TITLE
Refactor `summarize_change()`

### LINKED ISSUE #1345: refactor `summarize_change` function,
Rationale:

Issue was identified via updating the lbt03 template in scda.test using pharamverseadam data, 

https://github.com/insightsengineering/scda.test/pull/153/files#diff-612b47e07d81651ea501f536e8b1e59b19e5018ba14f7cfe1efe13544df8620c

zero was reported, after some investigation, we found out in recent admiral, there was a change to set CHG to NA for baseline values, discussion was tracked in https://github.com/pharmaverse/admiral/issues/2484

For lbt03 table, the summary statistics should be based on baseline AVAL values for baseline section, but for other visits in AVISIT, use CHG values, baseline values are flagged via baseline_flag: ABLFL

TODO:

1. update summarize_change function
2. update a_change_from_baseline function, better handle NA, can preprocess them to zeros
3. adding test cases, allow CHG input to be NA

### PR DESCRIPTION
**Pull request by Melkiades**
Fixes #1345

### CODE REVIEW

File: R/summarize_change.R [hunk 1]
```diff
@@ -57,26 +58,41 @@ s_change_from_baseline <- function(df,
...
+  # Adding automatically extra parameters to the statistic function (see ?rtables::additional_fun_params)
+  extra_afun_params <- names(get_additional_analysis_fun_parameters(add_alt_df = FALSE))
+  x_stats <- do.call(
+    s_change_from_baseline,
+    args = c(
+      df = list(df),
+      retrieve_extra_afun_params(extra_afun_params),
+      list(...)
+    )
   )
```
Comment (Melkiades) [inline R/summarize_change.R:76]: we can add alt_df handling to the whole stack

File: R/summarize_change.R [hunk 2]
```diff
@@ -119,31 +135,61 @@ a_change_from_baseline <- make_afun(
...
+                             ),
+                             .labels = c(
+                               mean_sd = "Mean (SD)",
+                               mean_se = "Mean (SE)",
+                               median = "Median",
+                               range = "Min - Max"
+                             ),
                              .indent_mods = NULL) {
-  extra_args <- list(variables = variables, ...)
+  checkmate::assert_string(vars)
```
Comment (Melkiades) [inline R/summarize_change.R:164]: not fully necessary as with get_additional_analysis_fun_parameters() we get `.var`

File: vignettes/tern_functions_guide.Rmd [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+Adding a custom statistic (WIP)
+
+
+## For Developers
+
+In all of these layers there are specific parameters that need to be available, and, while `rtables` has multiple way to handle formatting and `NA` values, we had to decide how to correctly handle these and additional extra arguments. We follow the following scheme:
+
+Level 1: `summarize_change()`: all parameters without a starting dot `.*` are used or added to `extra_args`. Specifically, here we solve `NA` values by using `inclNAs` option in `rtables::analyze()`. This will add to `...` `na.rm = inclNAs`. Also `na_str` is here set. We may want to be statistic dependend in the future, but we still need to think how to accomplish that. We add the `rtables::additional_fun_params` to the analysis function so to make them available as `...` in the next level.
+
+Level 2: `a_change_from_baseline()`: all parameters starting with a dot `.` are used. Mainly `.stats`, `.formats`, `.labels`, and `.indent_mods` are used. We also add `extra_afun_params` to the `...` list for the statistical function. Notice the handling for additional parameters in the `do.call()` function.
```
Comment (Melkiades) [inline vignettes/tern_functions_guide.Rmd:123]: needs refinement

**Comment by pawelru** [comment]
There is a bug in R CMD CHECK workflow logic. Sorry about that. There is a PR to fix it: https://github.com/insightsengineering/r.pkg.template/pull/261

**Comment by Melkiades** [comment]
@pawelru this is the current error:
! Failed to render 'vignettes/tern_functions_guide.Rmd'.
✖ Quitting from lines 65-87 [unnamed-chunk-3] (tern_functions_guide.Rmd)
Caused by error:
! could not find function "get_additional_analysis_fun_parameters"

In the vignette this function is not present though. Am I missing something obvious?

File: R/utils_rtables.R [hunk 4]
```diff
@@ -467,3 +467,68 @@ set_default_na_str <- function(na_str) {
...
+}
+
+#' @describeIn util_handling_additional_fun_params Curated list of additional parameters for
+#'   analysis functions. Please check [rtables::additional_fun_params] for precise descriptions.
+#'
+#' @return
+#' * `get_additional_analysis_fun_parameters` returns a list of additional parameters.
+#'
+#' @keywords internal
+get_additional_afun_params <- function(add_alt_df = FALSE) {
```
Comment (shajoezhu) [inline R/utils_rtables.R:510]: get_additional_analysis_fun_parameters

get_additional_afun_params?

Comment (shajoezhu) [reply R/utils_rtables.R:510]: perhaps lets just use get_additional_afun_params

File: R/summarize_change.R [hunk 5]
```diff
@@ -119,31 +144,60 @@ a_change_from_baseline <- make_afun(
...
-    a_change_from_baseline,
-    .stats = .stats,
-    .formats = .formats,
-    .labels = .labels,
-    .indent_mods = .indent_mods
+  # Adding additional arguments to the analysis function (depends on the specific call)
+  extra_args <- c(extra_args, "variables" = list(variables), ...)
+
+  # Adding all additional information from layout to analysis functions (see ?rtables::additional_fun_params)
+  extra_args[[".additional_fun_parameters"]] <- get_additional_analysis_fun_parameters(add_alt_df = FALSE)
```
Comment (shajoezhu) [inline R/summarize_change.R:183]: get_additional_afun_params

**Comment by shajoezhu** [comment]
> @pawelru this is the current error:
> 
> ```
> 
> ! Failed to render 'vignettes/tern_functions_guide.Rmd'.
> ✖ Quitting from lines 65-87 [unnamed-chunk-3] (tern_functions_guide.Rmd)
> Caused by error:
> ! could not find function "get_additional_analysis_fun_parameters"
> ```
> 
> In the vignette this function is not present though. Am I missing something obvious?

it's the function name @Melkiades

**Comment by shajoezhu** [comment]
Error applying analysis function (var - CHG): could not find function ".split_default_from_custom_stats"
	occured at (row) path: AVISIT[Baseline]

**Comment by shajoezhu** [comment]
hi @Melkiades , can you please update the DESCRIPTION file

```diff
--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -144,6 +144,8 @@ Collate:
     'h_survival_duration_subgroups.R'
     'imputation_rule.R'
     'incidence_rate.R'
+    'individual_patient_plot.R'
+    'kaplan_meier_plot.R'
     'logistic_regression.R'
     'missing_data.R'
     'odds_ratio.R'
@@ -166,6 +168,7 @@ Collate:
     'summarize_glm_count.R'
     'summarize_num_patients.R'
     'summarize_patients_exposure_in_cols.R'
+    'summarize_variables.R'
```

and rerun roxygen

File: R/utils_rtables.R (hunk 4 above)
Comment (Melkiades) [reply R/utils_rtables.R:510]: it was the longer version but it is too long for lintr

Comment (Melkiades) [reply R/utils_rtables.R:510]: thanks

**Comment by Melkiades** [comment]
> hi @Melkiades , can you please update the DESCRIPTION file
> 
> ```diff
> --- a/DESCRIPTION
> +++ b/DESCRIPTION
> @@ -144,6 +144,8 @@ Collate:
>      'h_survival_duration_subgroups.R'
>      'imputation_rule.R'
>      'incidence_rate.R'
> +    'individual_patient_plot.R'
> +    'kaplan_meier_plot.R'
>      'logistic_regression.R'
>      'missing_data.R'
>      'odds_ratio.R'
> @@ -166,6 +168,7 @@ Collate:
>      'summarize_glm_count.R'
>      'summarize_num_patients.R'
>      'summarize_patients_exposure_in_cols.R'
> +    'summarize_variables.R'
> ```
> 
> and rerun roxygen

`roxygen2::update.collate(".")` does not apply any modifications in my local (there should not be any difference in source files afaik)

**Review by shajoezhu — CHANGES_REQUESTED** [review]
hi @Melkiades , the CI test is still not passing

can you take another look. thanks!

```
a_change_from_baseline: no visible binding for global variable
  ‘.df_row’
a_change_from_baseline: no visible binding for global variable ‘.var’
get_additional_afun_params: missing arguments not allowed in calls to
  ‘c’
Undefined global functions or variables:
  .df_row .var
```

**Comment by shajoezhu** [comment]
> > hi @Melkiades , can you please update the DESCRIPTION file
> > ```diff
> > --- a/DESCRIPTION
> > +++ b/DESCRIPTION
> > @@ -144,6 +144,8 @@ Collate:
> >      'h_survival_duration_subgroups.R'
> >      'imputation_rule.R'
> >      'incidence_rate.R'
> > +    'individual_patient_plot.R'
> > +    'kaplan_meier_plot.R'
> >      'logistic_regression.R'
> >      'missing_data.R'
> >      'odds_ratio.R'
> > @@ -166,6 +168,7 @@ Collate:
> >      'summarize_glm_count.R'
> >      'summarize_num_patients.R'
> >      'summarize_patients_exposure_in_cols.R'
> > +    'summarize_variables.R'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and rerun roxygen
> 
> `roxygen2::update.collate(".")` does not apply any modifications in my local (there should not be any difference in source files afaik)

weired. when I run locally, these get added, and after that, more document rd files get rendered.

**Comment by shajoezhu** [comment]
> > hi @Melkiades , can you please update the DESCRIPTION file
> > ```diff
> > --- a/DESCRIPTION
> > +++ b/DESCRIPTION
> > @@ -144,6 +144,8 @@ Collate:
> >      'h_survival_duration_subgroups.R'
> >      'imputation_rule.R'
> >      'incidence_rate.R'
> > +    'individual_patient_plot.R'
> > +    'kaplan_meier_plot.R'
> >      'logistic_regression.R'
> >      'missing_data.R'
> >      'odds_ratio.R'
> > @@ -166,6 +168,7 @@ Collate:
> >      'summarize_glm_count.R'
> >      'summarize_num_patients.R'
> >      'summarize_patients_exposure_in_cols.R'
> > +    'summarize_variables.R'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > and rerun roxygen
> 
> `roxygen2::update.collate(".")` does not apply any modifications in my local (there should not be any difference in source files afaik)

I think I know why now. I use `devtools::document(roclets = c('rd', 'collate', 'namespace'))`

**Comment by shajoezhu** [comment]
downstream works ok https://github.com/insightsengineering/scda.test/pull/161/files

**Comment by Melkiades** [comment]
> roxygen2::update_collate(".")

I do have that version but those files are not present in R/

**Review by shajoezhu — APPROVED** [review]
lgtm! Thanks @Melkiades

**Pull request merged**